clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -358,7 +358,7 @@ public void writeTo(ByteBuffer buffer) {
      * @throws SchemaException If validation fails
      */
     private void validateField(BoundField field) {
-        if (this.schema != field.schema)
+        if (!this.schema.equals(field.schema))
```

Comment:
Is this change actually needed? The intent currently is to let `BoundField` be constructed only within the scope of a given Schema and to use `Schema.get` in order to get an instance. We do not expect a `BoundField` constructed in the context of one schema to be used for another schema.

Note also that the includes test case does not cover this change, so we are probably missing another test case if we think this fix is important.

Reply:
I originally thought that this change was needed since two schemas were being compared using the not equal operator, but I did not know the context in which the comparison was made.
Hence, that was why I added it.